ambry-frontend/src/main/java/com.github.ambry.frontend/PostBlobHandler.java

```diff
@@ -200,5 +195,85 @@ private void start() {
+      if (blobProperties.isEncrypted()) {
+        restRequest.getMetricsTracker()
+            .injectMetrics(
+                frontendMetrics.postRequestMetricsGroup.getRestRequestMetrics(restRequest.isSslUsed(), true));
+      }
+      byte[] userMetadata = RestUtils.buildUserMetadata(restRequest.getArgs());
+      frontendMetrics.blobPropsBuildTimeInMs.update(System.currentTimeMillis() - propsBuildStartTime);
+      LOGGER.trace("Blob properties of blob being POSTed - {}", blobProperties);
+      return new BlobInfo(blobProperties, userMetadata);
+    }
+
+    /**
+     * @return the {@link PutBlobOptions} to use, parsed from the request.
+     */
+    private PutBlobOptions getPutBlobOptionsFromRequest() throws RestServiceException {
+      PutBlobOptionsBuilder builder =
+          new PutBlobOptionsBuilder().chunkUpload(RestUtils.isChunkUpload(restRequest.getArgs()));
+      Long maxUploadSize = RestUtils.getLongHeader(restRequest.getArgs(), RestUtils.Headers.MAX_UPLOAD_SIZE, false);
+      if (maxUploadSize != null) {
+        builder.maxUploadSize(maxUploadSize);
+      }
+      return builder.build();
+    }
+
+    /**
+     * Attach the metadata to include in a signed ID to the {@link RestRequest} if the request is for a chunk upload.
+     * This will tell the ID converter that it needs to produce a signed ID to give back to the client.
+     * @param blobProperties the {@link BlobProperties} from the request.
+     * @throws RestServiceException
+     */
+    private void setSignedIdMetadataIfRequired(BlobProperties blobProperties) throws RestServiceException {
+      if (RestUtils.isChunkUpload(restRequest.getArgs())) {
+        Map<String, String> metadata = new HashMap<>(2);
+        metadata.put(RestUtils.Headers.BLOB_SIZE, Long.toString(restRequest.getBytesReceived()));
+        metadata.put(RestUtils.Headers.SESSION,
+            RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SESSION, true));
+        metadata.put(EXPIRATION_TIME_MS_KEY,
+            Long.toString(Utils.addSecondsToEpochTime(time.milliseconds(), blobProperties.getTimeToLiveInSeconds())));
+        restRequest.setArg(RestUtils.InternalKeys.SIGNED_ID_METADATA_KEY, metadata);
+      }
+    }
+
+    /**
+     * Enforce any additional requirements for certain types of uploads like data chunk uploads.
+     * @param blobProperties the {@link BlobProperties} parsed from the request.
+     * @throws RestServiceException
+     */
+    private void checkUploadRequirements(BlobProperties blobProperties) throws RestServiceException {
+      if (RestUtils.isChunkUpload(restRequest.getArgs())) {
+        // ensure that the x-ambry-session header is present.
+        RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SESSION, true);
+        // validate that max chunk size configuration is less than the chunking threshold.
+        long maxChunkSize = RestUtils.getLongHeader(restRequest.getArgs(), RestUtils.Headers.MAX_UPLOAD_SIZE, true);
```

Comment:
This is another place where `required` is set to true and the function should throw an exception instead of returning null.

If you think it's safer to make the check here as well instead of relying on the method implementation I can do so.